overlord/devicestate/handlers.go

```diff
@@ -85,7 +82,7 @@ func (m *DeviceManager) doGenerateDeviceKey(t *state.Task, _ *tomb.Tomb) error {
 		return nil
 	}
 
-	keyPair, err := rsa.GenerateKey(rand.Reader, keyLength)
+	keyPair, err := generateRSAKey(keyLength)
```

Comment:
this is something we should have done before, we can unlock around this:
st.Unlock()
keyPari, err := ...
st.Lock()